pyomo/core/expr/relational_expr.py

```diff
@@ -363,105 +397,319 @@ def _to_string(self, values, verbose, smap):
+    return RangedExpression((a,) + b.args, (True, b._strict))
+
+
+def _lt_ineq_expr(a, b):
+    return RangedExpression(a.args + (b,), (a._strict, True))
+
+
+def _lt_ineq_param(a, b):
+    if b.is_constant():
+        b = b.value
+    return RangedExpression(a.args + (b,), (a._strict, True))
+
+
+def _lt_param(a, b):
+    if a.is_constant():
+        a = a.value
+        if b.is_constant():
+            return a < b.value
+    elif b.is_constant():
+        b = b.value
+    return InequalityExpression((a, b), True)
+
+
+def _lt_param_any(a, b):
+    if a.is_constant():
+        return a.value < b
+    return InequalityExpression((a, b), True)
+
+
+def _lt_any_param(a, b):
+    if b.is_constant():
+        return a < b.value
+    return InequalityExpression((a, b), True)
+
+
+def _register_new_lt_handler(a, b):
+    types = _categorize_relational_arg_types(a, b)
+    # Retrieve the appropriate handler, record it in the main
+    # _lt_dispatcher dict (so this method is not called a second time for
+    # these types)
+    _lt_dispatcher[a.__class__, b.__class__] = handler = _lt_type_handler_mapping[types]
+    # Call the appropriate handler
+    return handler(a, b)
+
+
+_lt_dispatcher = collections.defaultdict(lambda: _register_new_lt_handler)
+_lt_type_handler_mapping = _binary_op_dispatcher_type_mapping(
+    _lt_dispatcher,
+    {
+        (ARG_TYPE.NATIVE, ARG_TYPE.NATIVE): _lt_native,
+        (ARG_TYPE.NATIVE, ARG_TYPE.PARAM): _lt_any_param,
+        (ARG_TYPE.NATIVE, ARG_TYPE.OTHER): _lt_expr,
+        (ARG_TYPE.NATIVE, ARG_TYPE.INEQUALITY): _lt_expr_ineq,
+        (ARG_TYPE.NATIVE, ARG_TYPE.INVALID_RELATIONAL): _lt_invalid,
+        (ARG_TYPE.PARAM, ARG_TYPE.NATIVE): _lt_param_any,
+        (ARG_TYPE.PARAM, ARG_TYPE.PARAM): _lt_param,
+        (ARG_TYPE.PARAM, ARG_TYPE.OTHER): _lt_param_any,
+        (ARG_TYPE.PARAM, ARG_TYPE.INEQUALITY): _lt_param_ineq,
+        (ARG_TYPE.PARAM, ARG_TYPE.INVALID_RELATIONAL): _lt_invalid,
+        (ARG_TYPE.OTHER, ARG_TYPE.NATIVE): _lt_expr,
+        (ARG_TYPE.OTHER, ARG_TYPE.PARAM): _lt_any_param,
+        (ARG_TYPE.OTHER, ARG_TYPE.OTHER): _lt_expr,
+        (ARG_TYPE.OTHER, ARG_TYPE.INEQUALITY): _lt_expr_ineq,
+        (ARG_TYPE.OTHER, ARG_TYPE.INVALID_RELATIONAL): _lt_invalid,
+        (ARG_TYPE.INEQUALITY, ARG_TYPE.NATIVE): _lt_ineq_expr,
+        (ARG_TYPE.INEQUALITY, ARG_TYPE.PARAM): _lt_ineq_param,
+        (ARG_TYPE.INEQUALITY, ARG_TYPE.OTHER): _lt_ineq_expr,
+        (ARG_TYPE.INEQUALITY, ARG_TYPE.INEQUALITY): _lt_invalid,
+        (ARG_TYPE.INEQUALITY, ARG_TYPE.INVALID_RELATIONAL): _lt_invalid,
+        (ARG_TYPE.INVALID_RELATIONAL, ARG_TYPE.NATIVE): _lt_invalid,
+        (ARG_TYPE.INVALID_RELATIONAL, ARG_TYPE.PARAM): _lt_invalid,
+        (ARG_TYPE.INVALID_RELATIONAL, ARG_TYPE.OTHER): _lt_invalid,
+        (ARG_TYPE.INVALID_RELATIONAL, ARG_TYPE.INEQUALITY): _lt_invalid,
+        (ARG_TYPE.INVALID_RELATIONAL, ARG_TYPE.INVALID_RELATIONAL): _lt_invalid,
+    },
+    ARG_TYPE,
+)
```

Comment:
I don't know if there is a "better" way to do this, but there are three versions of this map that are all "almost" the same, and it makes my little "minimize repeat code" heart sad.